generate_manifest.R

```diff
@@ -0,0 +1,29 @@
+#############
+# Get a manifest of files to upload
+#############
+
+#############
+### Synapse credentials and params
+#############
+# Set the environment variables .Renviron file in your home folder. Refer to README for more details
+SYNAPSE_AUTH_TOKEN = Sys.getenv('SYNAPSE_AUTH_TOKEN')
+SYNAPSECLIENT_INSTALL_PATH = Sys.getenv('SYNAPSECLIENT_INSTALL_PATH')
+source('~/recover-s3-synindex/params.R')
+### First run data_sync.R
+
+## The S3 bucket is synced to AWS_DOWNLOAD_LOCATION locally. 
+## We will use synapse cmd line client manifest function to replicate the folder structure 
+## but NOT upload the files. We will create a datafilehandleid in place later instead of uploading file
+
+old_path <- Sys.getenv("PATH")
+
+if(!grepl(SYNAPSECLIENT_INSTALL_PATH,old_path)){
+  Sys.setenv(PATH = paste(old_path, SYNAPSECLIENT_INSTALL_PATH, sep = ":"))
+  ## When we install synapser, it installs synapseclient and along with it the synapse cmd line client
+  ## We need to add the location of synapseclient to the system path so that it can recognize synapse cmd 
+}
+
+SYSTEM_COMMAND <- glue::glue('SYNAPSE_AUTH_TOKEN="{SYNAPSE_AUTH_TOKEN}" synapse manifest --parent-id {SYNAPSE_PARENT_ID} --manifest ./current_manifest.tsv {AWS_DOWNLOAD_LOCATION}')
```

Comment:
Have you tried it without the `SYNAPSE_AUTH_TOKEN={SYNAPSE_AUTH_TOKEN}` in the front?  I think the code _should work_ but I could be wrong.

Reply:
I think it would work without the `SYNAPSE_AUTH_TOKEN={SYNAPSE_AUTH_TOKEN}` in the front, but I went ahead and explicitly wrote that down so we know what SYNAPSE_AUTH_TOKEN we are gluing in before sending the command to the terminal.

Reply:
For security purposes, the issue here is that you run the risk of the auth token being exposed in the logs or stack trace which isn't great.

Reply:
Tom, I checked it does work without `SYNAPSE_AUTH_TOKEN={SYNAPSE_AUTH_TOKEN}` in the front, I made the changes to the code to reflect that. Thanks :)